src/lightmap.cpp

```diff
@@ -331,6 +331,15 @@ void map::generate_lightmap()
                 }
             }
         }
+        // atomic light is always on
+        std::vector<int> light_indices = v->all_parts_with_feature(VPFLAG_ATOMIC_LIGHT);
```

Comment:
For the initial check I can see `savegame_json.cpp`, however updating the flag may do well if the vehicle is marked "dirty" somehow, as I'm sure there is vehicle caching code elsewhere.